src/protocol/tx_format/input.md

```diff
@@ -78,20 +78,21 @@ Transaction is invalid if:
 
 ## InputMessage
 
-| name                  | type       | description                                                  |
-|-----------------------|------------|--------------------------------------------------------------|
-| `messageID`           | `byte[32]` | The messageID as described [here](../id/utxo.md#message-id). |
-| `sender`              | `byte[32]` | The address of the message sender.                           |
-| `recipient`           | `byte[32]` | The address or predicate root of the message recipient.      |
-| `amount`              | `uint64`   | Amount of base asset coins sent with message.                |
-| `nonce`               | `uint64`   | The message nonce.                                           |
-| `witnessIndex`        | `uint8`    | Index of witness that authorizes spending the coin.          |
-| `dataLength`          | `uint16`   | Length of message data, in bytes.                            |
-| `predicateLength`     | `uint16`   | Length of predicate, in instructions.                        |
-| `predicateDataLength` | `uint16`   | Length of predicate input data, in bytes.                    |
-| `data`                | `byte[]`   | The message data.                                            |
-| `predicate`           | `byte[]`   | Predicate bytecode.                                          |
-| `predicateData`       | `byte[]`   | Predicate input data (parameters).                           |
+| name                  | type                                   | description                                                  |
+|-----------------------|----------------------------------------|--------------------------------------------------------------|
+| `messageID`           | `byte[32]`                             | The messageID as described [here](../id/utxo.md#message-id). |
```

Comment:
I know it was here before, but why must we duplicate the `message_id` field if we can compute it based on the other fields?

<img width="527" alt="image" src="https://user-images.githubusercontent.com/18346821/220890194-547e7b1e-bd0b-464d-a11e-87c7b670caa8.png">

I have only one assumption: because we want to put it into the memory of the `vm` to be able to read it, instead of each time calculation.

But does this optimization worth 32 bytes per input?

@adlerjohn @Voxelot

Reply:
Hmm it does seem like we could omit the message id on inputs, since we have to re-compute the hash to validate it before spending anyways.

I'm also not sure how fraud-proofs would verify the messages. Since we can't lookup old events / logs, unless we put the message id into storage I'm not sure which fields are actually needed. cc @pixelcircuits

Reply:
So, we may end up putting the messageIds into storage for simplicity, but it's actually possible to prove emitted events given a starting block hash (you navigate down the chain and provide the necessary merkle proofs and hash preimages as necessary to show the log that occurred). I know Optimism intends to use this approach with Bedrock (essential complicate the fault proving, but optimizing for the optimistic case). This can also potentially be done in a zkProof via something like [Herodotus](https://www.herodotus.dev/) (their main focus is to provide proofs of other chain history, but it could be used to prove Ethereum history on Ethereum itself). It would actually be really cool to use zkProofs like Herodotus to prove output messages in our Fuel chain instead of having a list of output messages always copied into the block header

Reply:
@pixelcircuits  So, what is final conclusion?=) Do we need to keep the message id in the transaction, or it is okay to compute it during fraud proofs because we have access to all fields?